codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/DirectedRubyCodegen.java

```diff
@@ -98,8 +103,11 @@ public GenerationContext createContext(CreateContextDirective<RubySettings, Ruby
             protocolGenerator,
             applicationTransport,
             collectDependencies(model, service, protocol, directive.settings(), integrations),
-            directive.symbolProvider()
-        );
+            directive.symbolProvider());
+
+        typesGenerator = new TypesGenerator(context);
```

Comment:
I don't understand this one. Why is types generator special and why is it constructed here? With our old approach (orchestrator) the ordering of code generation was clear.

Reply:
Fixed